### PR TITLE
Add GLM MOJO compatibility tests, add GLM to mojoland init

### PR DESCRIPTION
**Pull request by km2018**
JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8041

This pull request adds MOJO compatibility checks for GLM.

### CODE REVIEW

File: mojo-py/mojoland/recipes/glm/cars_glm_recipe.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from ..datasets import cars_frame, cars_data
+from ..baserecipe import BaseRecipe
+
+
+class CarsGLMRecipe(BaseRecipe):
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = cars_frame()
+        model = H2OGeneralizedLinearEstimator(epochs=50)
+        model.train(y="mpg", training_frame=fr, ignored_columns=["name"])
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/cars_glm_recipe.py:15]: Karthik: remove the argument epochs=50, that one is for deeplearning.

File: mojo-py/mojoland/recipes/glm/eyestate_glm_recipe.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from ..baserecipe import BaseRecipe
+
+
+class EyestateGLMRecipe(BaseRecipe):
+    """Binomial model, all features are numeric."""
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = eyestate_frame()
+        model = H2OGeneralizedLinearEstimator(epochs=100, reproducible=True)
+        model.train(y="eyeDetection", training_frame=fr)
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/eyestate_glm_recipe.py:16]: Remove epochs=100, reproduci ble=True arguments.

File: mojo-py/mojoland/recipes/glm/iris_glm_recipe.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+from ..baserecipe import BaseRecipe
+
+
+class IrisGLMRecipe(BaseRecipe):
+    """Multinomial classification (3 classes), features are numeric (4 factors)."""
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = iris_frame()
+        model = H2OGeneralizedLinearEstimator(epochs=20, reproducible=True)
+        model.train(y="Species", training_frame=fr)
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/iris_glm_recipe.py:16]: remove arguments epochs=20, reproducible=True

File: mojo-py/mojoland/recipes/glm/missing_glm_recipe.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+from ..baserecipe import BaseRecipe
+
+
+class MissingGLMRecipe(BaseRecipe):
+    """Artificial dataset with missing values."""
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = missing_frame()
+        model = H2OGeneralizedLinearEstimator(epochs=50, reproducible=True)
+        model.train(training_frame=fr)
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/missing_glm_recipe.py:16]: remove epochs=50, reproducible=True

File: mojo-py/mojoland/recipes/glm/names_glm_recipe.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+class NamesGLMRecipe(BaseRecipe):
+    """Binomial classification with a categorical levels with 1575 factors."""
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = names_frame()
+        fr = fr[:5000, :]
+        fr["name"] = fr["name"].ascharacter().asfactor()  # trim nlevels()
+        assert 256 < fr["name"].nlevels()[0] < 500
+        model = H2OGeneralizedLinearEstimator(epochs=100, reproducible=True)
+        model.train(y="sex", training_frame=fr)
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/names_glm_recipe.py:19]: remove epochs=100, reproducible=True

File: mojo-py/mojoland/recipes/glm/stars_glm_recipe.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class StarsGLMRecipe(BaseRecipe):
+    """Regression model, some input columns are ignored."""
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = stars_frame()
+        assert fr.type("distance") == "int"
+        model = H2OGeneralizedLinearEstimator(epochs=100, reproducible=True)
+        model.train(y="distance", training_frame=fr, ignored_columns=["name1", "name2"])
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/stars_glm_recipe.py:17]: remove arguments epochs=100, reproducible=True

File: mojo-py/mojoland/recipes/glm/titanic_glm_recipe.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class TitanicGLMRecipe(BaseRecipe):
+    """Multinomial classification (8 classes), with mixed feature types."""
+
+    def bake(self) -> H2OGeneralizedLinearEstimator:
+        fr = titanic_frame()
+        fr["parch"] = fr["parch"].asfactor()
+        model = H2OGeneralizedLinearEstimator(epochs=50, reproducible=True)
+        model.train(y="parch", training_frame=fr, ignored_columns=["name", "ticket", "boat", "home.dest"])
```
Comment (wendycwong) [inline mojo-py/mojoland/recipes/glm/titanic_glm_recipe.py:17]: remove epochs=50, reproducible=True

**Review by wendycwong — CHANGES_REQUESTED** [review]

File: mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_multiparams.txt [hunk 8]
```diff
@@ -0,0 +1,44 @@
+2
```
Comment (koniecsveta) [inline mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_multiparams.txt:1]: is mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars.mojo really empty file?

**Comment by koniecsveta** [comment]
to me something seems to be wrong. when I run `python mojo-py/in_mojo_veritas.py --filter glm ` I get this exception:


```
...

#----------------------------------------------------------
#  Tasting Glm Cars recipe
#----------------------------------------------------------

Tasting nibble parameters in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_parameters.txt... ok
Tasting nibble multiparams in /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_multiparams.txt... ok
Traceback (most recent call last):
  File "mojo-py/in_mojo_veritas.py", line 40, in <module>
    connoisseur.degustate(recipe)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 39, in degustate
    self._taste(recipe, version)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 150, in _taste
    nibble_fresh = self._make_nibble(mojo, commands)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 229, in _make_nibble
    res = mojo.call(*command)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/mojo_model.py", line 27, in call
    return self._backend.invoke_method(self._id, method, params)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/backend/mojobackend.py", line 39, in invoke_method
    return self._request("GET /mojos/%s/%s" % (model_id, method), params=params)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/backend/mojobackend.py", line 162, in _request
    raise Exception("Error %d: %s\n>> Request: %s\n>> Params:  %r" %
Exception: Error 500: java.lang.IllegalAccessException: Class ai.h2o.mojos.server.core.MojoApi$ApiMethod can not access a member of class hex.genmodel.algos.glm.GlmMojoModelBase with modifiers "public final"

ai.h2o.mojos.server.core.MojoApi$ApiMethod.invoke(MojoApi.java:59)
ai.h2o.mojos.server.handlers.MojoApiHandler.executeModelMethod(MojoApiHandler.java:116)
ai.h2o.mojos.server.handlers.MojoApiHandler.getImpl(MojoApiHandler.java:53)
ai.h2o.mojos.server.handlers.BaseHandler.doGet(BaseHandler.java:20)

>> Request: GET /mojos/1/score0~dada
>> Params:  {'arg1': '[8,360,175,3821,11,73]', 'arg2': '[0, 0]'}
Closing connection _sid_bbf9 at exit
H2O session _sid_bbf9 closed.


```
while for all the other algos everything works ok.

and I get the same when I try to generate the .mojos and .txts from scratch `python mojo-py/in_mojo_veritas.py --filter glm --creative`:




```
...

#----------------------------------------------------------
#  Tasting Glm Cars recipe
#----------------------------------------------------------

Baking the recipe...
Parse progress: |█████████████████████████████████████████████████████████████████████████████| 100%
glm Model Build progress: |███████████████████████████████████████████████████████████████████| 100%

Saving the mojo to /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars.mojo

Making nibble parameters -> /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_parameters.txt
Making nibble multiparams -> /Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-data/mojos/glm/cars/v1.00/glm_v1.00_cars_multiparams.txt
Traceback (most recent call last):
  File "mojo-py/in_mojo_veritas.py", line 40, in <module>
    connoisseur.degustate(recipe)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 39, in degustate
    self._taste(recipe, version)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 150, in _taste
    nibble_fresh = self._make_nibble(mojo, commands)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/recipes/connoisseur.py", line 229, in _make_nibble
    res = mojo.call(*command)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/mojo_model.py", line 27, in call
    return self._backend.invoke_method(self._id, method, params)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/backend/mojobackend.py", line 39, in invoke_method
    return self._request("GET /mojos/%s/%s" % (model_id, method), params=params)
  File "/Users/zuzanaolajcova/IdeaProjects/mojoland/mojo-py/mojoland/backend/mojobackend.py", line 162, in _request
    raise Exception("Error %d: %s\n>> Request: %s\n>> Params:  %r" %
Exception: Error 500: java.lang.IllegalAccessException: Class ai.h2o.mojos.server.core.MojoApi$ApiMethod can not access a member of class hex.genmodel.algos.glm.GlmMojoModelBase with modifiers "public final"

ai.h2o.mojos.server.core.MojoApi$ApiMethod.invoke(MojoApi.java:59)
ai.h2o.mojos.server.handlers.MojoApiHandler.executeModelMethod(MojoApiHandler.java:116)
ai.h2o.mojos.server.handlers.MojoApiHandler.getImpl(MojoApiHandler.java:53)
ai.h2o.mojos.server.handlers.BaseHandler.doGet(BaseHandler.java:20)

>> Request: GET /mojos/1/score0~dada
>> Params:  {'arg1': '[8,360,175,3821,11,73]', 'arg2': '[0, 0]'}
Closing connection _sid_8044 at exit
H2O session _sid_8044 closed.

```

Let's agree on call to compare setups and workflow.

**Comment by michalkurka** [comment]
@km2018 please remove all the `.DS_Store` files, they are not needed

**Review by koniecsveta — APPROVED** [review]

**Pull request merged**